singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java

```diff
@@ -252,6 +259,16 @@ boolean isLoggingAllowedByDecider() {
       if (map.containsKey(logDecider)) {
         result = map.get(logDecider) != 0;
       }
+      if (result && disableDeciders != null) {
+        for (String disableDecider : disableDeciders) {
+          if (map.containsKey(disableDecider) && map.get(disableDecider) == 100) {
+            LOG.info("Disabling log stream {} because fleet disable decider {} is set to 100",
```

Comment:
For tracking purposes would it help if we emit a metric here (maybe only the first time it is detected or when it changes so we don't overwhelm metric agent)?

Reply:
We can, I can probably add it right before we return here so that its emitting the metric for as long as the disable decider is turned on (which ideally shouldn't be for long)